### ISSUE TITLE
Missing remainging time under openCL

### ISSUE THREAD

**Issue by tuxun** [laurent-clouet/sheepit-client#279]
Hi, here is a work in progress Proposal DRAFT for fix the missing remaining time shown on client when we use opencl

At the moment, in Line 211 of Job.java:
```java
		if (isUseGPU() && configuration.getGPUDevice() != null && configuration.getComputeMethod() != ComputeType.CPU && OpenCL.TYPE
				.equals(configuration.getGPUDevice().getType())) {
			new_env.put("CYCLES_OPENCL_SPLIT_KERNEL_TEST", "1");
			this.updateRenderingStatusMethod = UPDATE_METHOD_BY_TILE; // don't display remaining time
```
I suggest 
```java
		if (isUseGPU() && configuration.getGPUDevice() != null && configuration.getComputeMethod() != ComputeType.CPU && OpenCL.TYPE
				.equals(configuration.getGPUDevice().getType())) {
			new_env.put("CYCLES_OPENCL_SPLIT_KERNEL_TEST", "1");
			// how we find remaining time in "console log" for show it in the UI: (using updateRenderingStatus(String) below)
			this.updateRenderingStatusMethod = UPDATE_METHOD_BY_REMAINING_TIME;
```
result:  time remaining with open cl is shown.

[![sheepitremaningtime.png](https://i.postimg.cc/dtgTtKZq/sheepitremaningtime.png)](https://postimg.cc/QHkM0wqP)

please help me to fix the percentage or i'll finish  monday (i go in 4 day in week end).
also, I would factorize a bit and take the time to make a real PR

Have  nice weekends, everyone!

updated: code change proposal

**Comment by tuxun** [comment]
@laurent-clouet 
hi,  something is bothering me, what was the specific reason for "don't display remaining time"?
```
if (isUseGPU() && configuration.getGPUDevice() != null && configuration.getComputeMethod() != ComputeType.CPU && OpenCL.TYPE
				.equals(configuration.getGPUDevice().getType())) {
			new_env.put("CYCLES_OPENCL_SPLIT_KERNEL_TEST", "1");
			this.updateRenderingStatusMethod = UPDATE_METHOD_BY_TILE; // don't display remaining time
		}
```
at start? (job.java, lines 211->214 on master)

**Comment by Raimund58** [comment]
@luguina Could you help our friend here a little bit out? :)

**Comment by luguina** [comment]
> @luguina Could you help our friend here a little bit out? :)

Sure thing, mate! I'll check tonight as soon as I'm in front of the keyboard 😃

**Comment by tuxun** [comment]
Afaik, I commented the line putting remaining shown time in UPDATE_METHOD_BY_TILE mode when we use OPENCL under linux and the remaining time brought back again in ui.
I understand the return from the log of Blender is not very linear, which can be a problem for make the percentage based progress bar in the taskbar icon, and the percentage shown in "status" UI field.

I had no reason to modifiy the case UPDATE_METHOD_BY_REMAINING_TIME if the method UPDATE_METHOD_BY_TILE works

It could be annoying for you to test if it works cause OPENCL rendering is disabled on linux since #165

**Comment by tuxun** [comment]
Tested on two machines of two different account 
![](https://cdn.discordapp.com/attachments/301912419368697856/735651283570262107/Clipboard01.jpg)

![](https://cdn.discordapp.com/attachments/735433986460876841/735758813847617567/unknown.png)

**Comment by tuxun** [comment]
fixed with #282

**Comment by tuxun** [comment]
note: if you still encounter a missing remaining time, it will PROBABLY be because a "old" loaded rendering kernel isn't logging the remaining time in the same manner than the *normal/updated* ones.
